tests/output/ConsoleInteractionTest.py

```diff
@@ -183,6 +186,68 @@ def test_require_settings(self):
 
             self.assertEqual(generator.last_input, 2)
 
+    def test_print_diffs_info(self):
+        file_dict = {"a": ["a\n", "b\n", "c\n"], "b": ["old_first\n"]}
+        diff_dict = {"a": Diff(file_dict['a']),
+                     "b": Diff(file_dict['b'])}
+        diff_dict["a"].add_lines(1, ["test\n"])
+        diff_dict["a"].delete_line(3)
+        diff_dict["b"].add_lines(0, ["first\n"])
+        previous_diffs = {"a": Diff(file_dict['a'])}
+        previous_diffs["a"].change_line(2, "b\n", "b_changed\n")
+        with retrieve_stdout() as stdout:
+            print_diffs_info(diff_dict,
+                             file_dict,
+                             previous_diffs,
+                             self.console_printer),
+            self.assertEqual(stdout.getvalue(),
+                             "|    | +1 -1 in a\n"
+                             "|    | +1 -0 in b\n")
```

Comment:
ok

Sincerely,

Lasse Schuirmann

contact@viperdev.io
http://viperdev.io/

2016-05-06 16:10 GMT+02:00 Attila <notifications@github.com>:

> In tests/output/ConsoleInteractionTest.py
> <https://github.com/coala-analyzer/coala/pull/2113#discussion_r62334362>:
>
> > +        file_dict = {"a": ["a\n", "b\n", "c\n"], "b": ["old_first\n"]}
> > +        diff_dict = {"a": Diff(file_dict['a']),
> > +                     "b": Diff(file_dict['b'])}
> > +        diff_dict["a"].add_lines(1, ["test\n"])
> > +        diff_dict["a"].delete_line(3)
> > +        diff_dict["b"].add_lines(0, ["first\n"])
> > +        previous_diffs = {"a": Diff(file_dict['a'])}
> > +        previous_diffs["a"].change_line(2, "b\n", "b_changed\n")
> > +        with retrieve_stdout() as stdout:
> > +            print_diffs_info(diff_dict,
> > +                             file_dict,
> > +                             previous_diffs,
> > +                             self.console_printer),
> > +            self.assertEqual(stdout.getvalue(),
> > +                             "|    | +1 -1 in a\n"
> > +                             "|    | +1 -0 in b\n")
>
> I believe that tests should be as separate as possible and it's not big
> deal if there is repetitions in them.
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/coala-analyzer/coala/pull/2113/files/eba0b25381f719fe3a373b78e8522401d7dbc863..459000e955485a38a41272dc0748fe65e0204837#r62334362>
>